Add testing guidance to the contributing guide (#5808)
grommet/grommet: CONTRIBUTING.md

## Testing Your Code

> - You can test your code locally along with your changes using: `yarn storybook` or `npm run storybook`. This will open the storybook in your browser.

Reviewer: ```suggestion
**Manual Tests**
- You can test your code locally along with your changes using: `yarn storybook` or `npm run storybook`. This will open the storybook in your browser.
```

> This project contains Jest tests. These tests can be run using `yarn test`. When new features are added, a Jest test should also be added. The following best practices should be observed when writing Jest tests:

Reviewer 2: ...`yarn test` or `npm run test`

Reviewer 2: "When new features are added..." -> "When changes to functionality are made, tests should always be included". Something along those lines could be more clear.

Reviewer 2: Maybe to be more clear about testing and what technologies are used ... "This project contains test that are executed using `jest`. The bulk of the component tests are written with React Testing Library..."

Author: Good comments! I included these in the latest commit and made it more clear that I was referring to react testing library

> - `screen` should be used for querying.

Reviewer 2: When the reader gets to this point, it would be helpful to know that React Testing Library is what is being referred to here.

Reviewer 2: Also if theres an example of a "good" test, it might be helpful to link to it here.

Author: In latest commit I clarified that the rules/guidelines are related to react testing library.

As far as including examples of good tests I was thinking of potentially going through a smaller component and re-writing the tests for it so they all conform with the best practices and then we could reference that whole test file as an example? For now the Common mistakes with React Testing Library article includes examples. But I agree it would be nice to have grommet examples.

> This project contains tests that are executed using Jest. The bulk of the component tests are written with React Testing Library. These tests can be run using `yarn test` or `npm run test`. When changes to functionality are made, tests should always be included. The following best practices should be observed when writing Jest tests with React Testing Library:

Reviewer: ```suggestion
This project contains unit tests executed by Jest. The bulk of the component tests are written with [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/) to simulate end user behavior and focus on testing functionality instead of implementation.

Jest tests are run using `yarn test` or `npm run test`. When changes to functionality are made, tests should always be included covering the functionality introduced.

The following best practices should be observed when writing Jest tests with React Testing Library:
```

Reviewer: ```suggestion
**Automated Tests**
This project contains tests that are executed using Jest. The bulk of the component tests are written with React Testing Library. These tests can be run using `yarn test` or `npm run test`. When changes to functionality are made, tests should always be included. The following best practices should be observed when writing Jest tests with React Testing Library:
```
